uitools/cpp/Esri/ArcGISRuntime/Toolkit/SearchViewController.cpp

```diff
@@ -0,0 +1,821 @@
+
+    Selecting a result will trigger changing the viewpoint of the geoView to that result
+    if applicable.
+   */
+  void SearchViewController::setSelectedResult(SearchResult* selectedResult)
+  {
+    if (m_selectedResult == selectedResult)
+      return;
+
+    if (m_selectedResult)
+      m_selectedResult->deleteLater();
+
+    if (m_graphicsOverlay)
+      m_graphicsOverlay->graphics()->clear();
+
+    m_selectedResult = selectedResult;
+
+    if (m_selectedResult)
+    {
+      // The SearchViewController takes control of the result, and we are going to delete the
+      // remaining results from the model as there's no need to keep them hanging about any further.
+      m_selectedResult->setParent(this);
+
+      connect(m_selectedResult, &QObject::destroyed, this, &SearchViewController::selectedResultChanged);
+
+      addGeoElementToOverlay(m_graphicsOverlay, m_selectedResult->geoElement());
+
+      if (auto sceneView = qobject_cast<SceneViewToolkit*>(m_geoView))
+      {
+        // Set sceneView viewpoint to where graphic is.
+        sceneView->setViewpoint(m_selectedResult->selectionViewpoint(), 0);
+      }
+      else if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+      {
+        // Set mapView callout and zoom to where graphic + callout are (if applicable.)
+        mapView->calloutData()->setTitle(m_selectedResult->displayTitle());
+        mapView->calloutData()->setDetail(m_selectedResult->displaySubtitle());
+        mapView->calloutData()->setGeoElement(m_selectedResult->geoElement());
+        mapView->calloutData()->setVisible(true);
+
+        mapView->setViewpoint(m_selectedResult->selectionViewpoint(), 0);
+      }
+    }
+    else
+    {
+      // Hide callout if we are unsetting the selectedResult.
+      if (auto mapView = qobject_cast<MapViewToolkit*>(m_geoView))
+      {
+        mapView->calloutData()->setVisible(false);
+      }
+    }
+
+    // Now that we own the selected result, we can clear the old ones out.
+    m_results->clear();
+
+    emit selectedResultChanged();
+  }
+
+  /*!
+      \brief Returns the collection of search sources to be used.
+   */
+  GenericListModel* SearchViewController::sources() const
+  {
+    return m_sources;
+  }
+
+  /*!
+      \brief Returns the collection of suggestion results from a query.
+   */
+  GenericListModel* SearchViewController::suggestions() const
+  {
+    return m_suggestions;
+  }
+
+  /*!
+      \brief Returns whether the user can perform a requery.
+      A requery will perform the search at the same location, but will only look at results
+      that exist within the current exist, ignoring all other results.
+   */
+  bool SearchViewController::isEligableForRequery() const
+  {
+    return m_isEligableForRequery;
+  }
+
+  /*!
+      \brief Sets whether the user can perform a requery to \a isEligableForRequery.
+      When \l automaticConfigurationEnabled is \c true, and the user moves the extent, this is automatically set to
```

Comment:
Optional, but perhaps "changes the viewpoint" rather than "moves the extent". I will leave final wording to you and Lucas.